### DISCUSSION TITLE
Tips for automatically update a package?

### DISCUSSION THREAD

**Discussion by De-Panther** [openupm/openupm#1727] [Q&A] [answered]
Sorry if it's not the place for this question.

I'm looking for a way to automatically update all the necessary files, and tag a commit, so OpenUPM would get the package with no errors.

I need to:
- Update `package.json` to the new version number.
- Update CHANGELOG file, move all the `Unreleased` updates to the new version.
- Create a release tag with the new version number, and text from the CHANGELOG.

And all should be in the same commit.

Any GitHub bot/action that can do that?
What other repositories maintainers does to make sure not to miss one of the update steps?

Thanks

[a comment by an automated account is not shown]

**Comment by favoyang** [comment, marked as answer]
Please checkout https://openupm.com/docs/managing-upm-project.html

Part 2 and Part 4.

**Comment by favoyang** [comment]
Live example: https://github.com/favoyang/unity-addressable-importer

**Comment by De-Panther** [comment]
Thanks!

It seems like what I need.
I do need to check if it works in repos that have more than one package.

**Comment by starikcetin** [comment]
@mob-sakai Is pretty good at this automatic release stuff. I think they use some sort of a semantic release runner hooked into GitHub actions in their repositories.
